src/rantanplan/core.py

```diff
@@ -98,6 +99,71 @@ def have_prosodic_liaison(first_syllable, second_syllable):
             and second_syllable['syllable'][0] in LIAISON_SECOND_PART)
 
 
+def get_synalephas(words):
+    prosodic_line = []
+    word_list = copy.deepcopy(words)
+    for idx, syllables_list in enumerate(word_list):
+        for syllable in syllables_list:
+            if not syllable.get('has_synalepha'):
+                prosodic_line.append(syllable)
+            else:
+                next_syllable = word_list[idx + 1].pop(0)
+                prosodic_line.append({'is_stressed': (syllable.get('is_stressed') or next_syllable.get('is_stressed')),
+                                      'syllable': "".join([syllable.get('syllable'), next_syllable.get('syllable')]),
+                                      'has_synalepha': True})
+    return prosodic_line
+
+
+def get_sinaeresis(word):
+    sinaeresis_words = word[:]  # create a copy since we're going to pop()
+    prosodic_syllables = []
+    for idx, key in enumerate(sinaeresis_words):
```

Comment:
Same thing, it feels like a proper for loop could avoid the use of the copy and the `pop`